apps/pollbook/backend/src/barcode_scanner/unix_socket.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  LogDispositionStandardTypes,
+  LogEventId,
+  Logger,
+} from '@votingworks/logging';
+import * as net from 'node:net';
+
+const UDS_PATH = '/tmp/barcodescannerd.sock';
+
+/*
+ * Attempts to connect to the barcode scanner Unix socket once. Resolves with
+ * the socket client if successful or rejects with an error.
+ */
+export function tryConnect(logger: Logger): Promise<net.Socket> {
+  return new Promise((resolve, reject) => {
+    const client = net.createConnection({
+      path: UDS_PATH,
+    });
+
+    client.once('connect', () => {
```

Comment:
Explicit binding makes it easier to test since we already need to support `client.once('error')` in `MockSocket`